src/actions/userActions.js

```diff
@@ -12,6 +16,27 @@ export const logoutSuccess = () => ({
   type: types.LOGOUT_SUCCESS
 });
 
+export const toggleLoading = () => ({
+  type: types.TOGGLE_LOADING
+});
+
+export const signUp = user =>
+  (dispatch) => {
+    dispatch(toggleLoading());
+    return userApi.signUp({ user }).then((response) => {
+      sessionService.saveUser(response)
+        .then(() => {
+          dispatch(toggleLoading());
```

Comment:
You can add the `signUpSuccess` action type to the reducer where you handle the loading, and turn it off there. It's good practice to dispatch as less actions as possible at any given time.

If you dispatch more actions, the reducers need to evaluate more times. Remember all reducers are listening for action types and you can write code for a certain type in more than one reducer.